pkg/controller/deployment/util/deployment_util_test.go

```diff
@@ -1020,6 +1020,68 @@ func TestMaxUnavailable(t *testing.T) {
 	}
 }
 
+func TestGetNonNegativeIntFromAnnotation(t *testing.T) {
+	tests := []struct {
+		name          string
+		annotations   map[string]string
+		expectedValue int32
+		expectedValid bool
+		expectErr     bool
+	}{
+		{
```

Comment:
Please add a test case for `max uint32`, and `max uint32 + 1` to verify the correct params are passed to ParseUInt

Reply:
Good idea, let's check the upper bound as well. I added a new check, but only for max int32, as we usually operate on these value only. And we never set a higher value to these annotations.

Reply:
for reference: 

https://github.com/kubernetes/kubernetes/blob/b845137026be136ec12d798a2b729ce8b99b85af/pkg/controller/deployment/util/deployment_util.go#L402

Reply:
sgtm